### PR TITLE
feat(dropdown): adjust menu position when there is not available space

### PR DESCRIPTION
**Pull request by SteliosKon**
## Description

This PR adjusts menu position when there is not available space

### CODE REVIEW

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -209,8 +214,8 @@ const ScrollableTemplate: Story<DropdownProps & React.CSSProperties> = ({
 
   return (
     <ScrollableContainer>
-      <div style={{ position, top }}>
-        <Dropdown {...args}>
+      <div style={{ position, top, marginInlineStart, height }}>
+        <Dropdown {...args} fixPlacement placement="top-end">
```
Comment (ThrasyvoulosKafasis) [inline src/components/Dropdown/Dropdown.stories.tsx:218]: `fixPlacement` and `placement` props overwrite story's controls and it's confusing that whatever you apply has no effect

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -26,10 +26,10 @@ const dropdownItemClasses = (item: DropdownItem): string =>
   });
 
 const fixPlacementMapping: Record<PlacementOptions, PlacementOptions> = {
```
Comment (ThrasyvoulosKafasis) [inline src/components/Dropdown/Dropdown.tsx:28]: `fixPlacementMapping` this var used for rtl direction mapping. Use a more appropriate name.

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -78,59 +81,106 @@ const Dropdown: FC<DropdownProps> = ({
...
+    const dropdownMenuWidth = dropdownWrapperRef.current.getBoundingClientRect().width;
 
-            setCurrentPlacement(spaceBelow < dropdownMenuHeight ? fixedPlacement : placement);
-          }
-          break;
+    let vertical = "bottom";
+    let horizontal = "start";
+
+    switch (placement) {
+      case "bottom-start": {
```
Comment (ThrasyvoulosKafasis) [inline src/components/Dropdown/Dropdown.tsx:95]: Add comments to explain the logic in each case

**Review by ThrasyvoulosKafasis — CHANGES_REQUESTED** [review]

File: src/components/Dropdown/Dropdown.stories.tsx (hunk 1 above)
Comment (SteliosKon) [reply src/components/Dropdown/Dropdown.stories.tsx:218]: Fixed [here](https://github.com/epignosis/gnosis/pull/300/commits/5bf1ddb15b587487deb02ae29efb68c9293178c8)

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (SteliosKon) [reply src/components/Dropdown/Dropdown.tsx:101]: Fixed [here](https://github.com/epignosis/gnosis/pull/300/commits/5bf1ddb15b587487deb02ae29efb68c9293178c8)

File: src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (SteliosKon) [reply src/components/Dropdown/Dropdown.tsx]: Fixed [here](https://github.com/epignosis/gnosis/pull/300/commits/5bf1ddb15b587487deb02ae29efb68c9293178c8)

File: src/components/Dropdown/Dropdown.stories.tsx (hunk 1 above)
Comment (ThrasyvoulosKafasis) [reply src/components/Dropdown/Dropdown.stories.tsx:218]: I can still see this issue. Fix placement is always applied as `true` and the placement is always the fixed one.

**Pull request merged**